pkg/infra/kvstore/kvstore.go

```diff
@@ -25,6 +25,7 @@ type KVStore interface {
 	Set(ctx context.Context, orgId int64, namespace string, key string, value string) error
 	Del(ctx context.Context, orgId int64, namespace string, key string) error
 	Keys(ctx context.Context, orgId int64, namespace string, keyPrefix string) ([]Key, error)
+	Items(ctx context.Context, orgId int64, namespace string) ([]Item, error)
```

Comment:
To match the above set of operations and using the [redis API as inspiration](https://redis.io/commands/hgetall/) I suggest two things:

1) `GetAll` as the function name on the interface
2) The return value to be a map of `map[string]Item` (assuming order does not matter)

Reply:
What is the string in the map in that case??

My first approach was to put `map[string]string` returning `key:value` but if you want to retrieve the Items from all organisations, the key couldn't be unique. It's why I returned an array instead a map.

Reply:
Good shout, I missed the organization bit. My suggestion was assuming `key -> item` in the map, but on a second look - I suggest:

`map[orgID]map[key]Item` (maybe even get rid of `Item` at this point in favour or pimitives) or `map[orgID][]Item` if you want to avoid the double map - I'd probably do the former because in my head when looping through all the items you'd be interested in all three of them within the same context: the two keys and the value .

Reply:
It was another idea that I had but at the beginning I thought that it was too:

![](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRWz2IllzqCAElLoPPxGA7vsM9XS3FoIW1WWg&usqp=CAU)

If we are going to use that I prefer the first one too. Let me update the PR.

Reply:
Finally I decided to use `string` instead of `Item`. `Item` has `orgId`, `namespace`, `key` and `value`. We have all values except the `value`, so I think that returning the whole `Item` not worth it.